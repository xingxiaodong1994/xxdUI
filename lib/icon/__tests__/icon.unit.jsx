// 测试icon组件
// 测试icon组件是否正常渲染。
import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'

describe('测试icon组件', () => {
  it('icon组件正常渲染', () => {
    const json=renderer.create(<Icon name='file' />).toJSON();
    expect(json).toMatchSnapshot()
  });
  it('icon组件onClick', () => {
    const fn=jest.fn();
    const c=mount(<Icon name='file' onClick={fn} />);
    c.find('svg').simulate('click');
    expect(fn).toBeCalled();
  })
});