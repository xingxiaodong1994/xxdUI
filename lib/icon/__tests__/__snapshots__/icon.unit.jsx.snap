// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试icon组件 icon组件正常渲染 1`] = `
<svg
  className="xxdui-icon"
>
  <use
    xlinkHref="#file"
  />
</svg>
`;
